### DISCUSSION TITLE
Support for bindings of other languanges (e.g. Python)

### DISCUSSION THREAD

**Discussion by cmpute** [microsoft/vcpkg#19065] [Ideas]
Many packages provides binding for python, java, etc. How about bring binding support into vcpkg? Take python for example, built python modules can be put into `installed/<triplet>/python/<port>` folder, and once `installed/<triplet>/python` is added to `PYTHON_PATH`, the modules are available in python. Adding to `PYTHON_PATH` can be a command of vcpkg as well.

Besides, I have a problem with python building. I'm trying to build `caffe` with python binding, however there's not a way to pass external python path to vcpkg. The python downloaded as a tool by vcpkg doesn't contain headers for building and the `python3` port has only python module. Python binding for caffe needs numpy. So I propose to add a option to pass external python path into vcpkg (or even for other languages).

**Comment by twMr7** [comment]
I was working on an idea as described below, but never got enough spare time to practice it.

Since there are so many python distributions, and anyone could have multiple version of python installed in local, I would like to suggest to use a meta package, e.g. `python3-dist`, which accept a port name as a feature to setup a specific distribution, e.g. `python3-dist-anaconda`.

All we need is to let other port be able to build with proper python environment, so a dedicate script for each distribution, e.g. `find_anaconda.cmake`, should be included in the distribution port, Any port depends on `pythons-dist` can then source the actual find_xxx.cmake script to setup required environment.

Well, I believe there is a more simpler way to do the trick, though.

**Comment by NancyLi1013** [comment]
This issue hasn’t been updated in a year; if it is still an issue, please reopen.

**Comment by nilsnolde** [comment]
@NancyLi1013 all issues I found to support external Python apps (#5514, #1974) were closed and there's no viable solution anywhere that I can see.

I think most packages having python(3) as dependency would benefit here. Esp boost-python currently will be compiled against whichever Python version is in `vcpkg`. When using boost-python to create Python bindings for a project (fairly common use case), at least in my project the compiled bindings are sensitive to the minor Python version. Meaning in `master` of vcpkg Python 3.8 is the standard, so boost-python will compile against that and I can **only** use those bindings with 3.8, **not 3.7, not 3.6**. 

How about re-opening one of these issues?

**Comment by petersteneteg** [comment]
This is most definitely still an issue. I have to manually remove the python dependency in pybind11 to be able to link and use the system python version. I for example need to have NumPy and many other python packages installed. And I don't want to have to install all of those in the python installation inside of vcpkg.
